Tasks/Grunt/grunttask.ts

```diff
@@ -41,11 +47,85 @@ gt.pathArg(gruntFile);
 
 gt.argString(tl.getInput('arguments', false));
 
-gt.exec()
-.then(function(code) {
-	tl.setResult(tl.TaskResult.Succeeded, tl.loc('GruntReturnCode', code));
-})
-.fail(function(err) {
+var enableCoverage = tl.getBoolInput('enableCodeCoverage');
+
+if (enableCoverage) {
+	var testSrc = tl.getPathInput('testFiles', true, false);
+
+	if (os.type().match(/^Win/)) {
+		var istanbul = tl.createToolRunner(tl.which('istanbul', true));
+	} else {
+		var istanbul = tl.createToolRunner(tl.which('./node_modules/istanbul/lib/cli.js', true));
+	}
+
+	istanbul.argString('cover --report cobertura');
+	istanbul.arg('./node_modules/mocha/bin/_mocha');
```

Comment:
are you sure? how does it know whether source file is covered by test case and which is source file? I hope you have covered this in your manual testing

---
In reply to: [66912485](https://github.com/Microsoft/vsts-tasks/pull/1880#discussion_r66912485) [](ancestors = 66912485)